tests/utils.go

```diff
@@ -469,6 +469,64 @@ func GetSupportedCPUModels(nodes k8sv1.NodeList) []string {
 	return cpus
 }
 
+func GetSupportedCPUFeaturesAmongAllTheNodes(nodes k8sv1.NodeList) []string {
+	var featureDenyList = map[string]bool{
+		"svm": true,
+	}
+	var supportedFeatures []string
+	for label := range nodes.Items[0].Labels {
+		if strings.Contains(label, services.NFD_CPU_FEATURE_PREFIX) {
+			feature := strings.TrimPrefix(label, services.NFD_CPU_FEATURE_PREFIX)
+			if _, ok := featureDenyList[feature]; !ok {
+				supportedFeatures = append(supportedFeatures, feature)
+			}
+		}
+	}
+
+	for _, node := range nodes.Items {
+		var currFeatures []string
+		for key := range node.Labels {
+			if strings.Contains(key, services.NFD_CPU_FEATURE_PREFIX) {
+				feature := strings.TrimPrefix(key, services.NFD_CPU_FEATURE_PREFIX)
+				if _, ok := featureDenyList[feature]; !ok {
+					currFeatures = append(currFeatures, feature)
+				}
+			}
+		}
+		supportedFeatures = setIntersection(supportedFeatures, currFeatures)
+	}
+
+	return supportedFeatures
+}
+func setIntersection(firstSet, secondSet []string) (intersection []string) {
+	firstSetBoolMap := make(map[string]bool)
+	var setOfFeaturesWithDups []string
+
+	for _, element := range firstSet {
+		firstSetBoolMap[element] = true
+
+	}
+
+	for _, element := range secondSet {
+		if firstSetBoolMap[element] {
+			setOfFeaturesWithDups = append(setOfFeaturesWithDups, element)
+		}
+	}
+	intersection = removeDups(setOfFeaturesWithDups)
+	return intersection
+}
+
+func removeDups(elements []string) (intersection []string) {
+	found := make(map[string]bool)
+	for _, element := range elements {
+		if found[element] == false {
```

Comment:
I'm trying to understand how can this ever happen ... afaiu, you never `map[key] = false` ... While I'm aware the default for a boolean is false, how can you add a key to a map without stating its value ? 

Furthermore, I think `if _, foundItem := found[element]; !foundItem { ... }` to be more golang idiomatic.

Reply:
Done